### ISSUE TITLE
Feature Request: Add Mish Activation

### ISSUE THREAD

**Issue by digantamisra98** [awslabs/djl#16]
Mish is a novel activation function proposed in this [paper](https://arxiv.org/abs/1908.08681).
It has shown promising results so far and has been adopted in several packages including:

|     |     |  |
| :---: | :---: |  :---: |
|[TensorFlow-Addons](https://github.com/tensorflow/addons/tree/master/tensorflow_addons/activations)|[SpaCy (Tok2Vec Layer)](https://github.com/explosion/spaCy)|[Thinc - SpaCy's official NLP based ML library](https://github.com/explosion/thinc/releases/tag/v7.3.0)|
|[Eclipse's deeplearning4j](https://github.com/eclipse/deeplearning4j/issues/8417)|[Hasktorch](https://github.com/hasktorch/hasktorch/blob/e5d4fb1b11663de3e032941f78b9d25c5da06f6d/hasktorch/src/Torch/Typed/Native.hs#L314)|[Echo AI](https://github.com/digantamisra98/Echo)|
|[CNTKX - Extension of Microsoft's CNTK](https://github.com/delzac/cntkx)|[FastAI-Dev](https://github.com/fastai/fastai_dev/blob/0f613ba3205990c83de9dba0c8798a9eec5452ce/dev/local/layers.py#L441)|[Darknet](https://github.com/AlexeyAB/darknet/commit/bf8ea4183dc265ac17f7c9d939dc815269f0a213)|
|[Yolov3](https://github.com/ultralytics/yolov3/commit/444a9f7099d4ff1aef12783704e3df9a8c3aa4b3)|[BeeDNN - Library in C++](https://github.com/edeforas/BeeDNN)|[Gen-EfficientNet-PyTorch](https://github.com/rwightman/gen-efficientnet-pytorch)|
|[dnet](https://github.com/umangjpatel/dnet)|[ruby-dnn](https://github.com/unagiootoro/ruby-dnn#implemented)|[blackcat-tensors](https://github.com/josephjaspers/blackcat_tensors/commit/87fa2f427d044a120816f2bc83176bdb455ceccd)|
|[DL4S](https://palle-k.github.io/DL4S/)|[HuggingFace Transformers](https://github.com/huggingface/transformers/pull/1753)|[PAGI](https://github.com/ProjectAGI/pagi)|
|[OpenCV](https://github.com/opencv/opencv/pull/15808)|[Odin-AI](https://github.com/imito/odin-ai)|[Mini DNN](https://github.com/yixuan/MiniDNN/commit/a405bd082183e685222472fa5851fd5024b1a7e6)|
|[Efficient Segmentation Networks](https://github.com/xiaoyufenfei/Efficient-Segmentation-Networks)|||

All benchmarks, analysis and links to official package implementations can be found in this [repository](https://github.com/digantamisra98/Mish) 

Mish also was recently used for a submission on the [Stanford DAWN Cifar-10 Training Time Benchmark](https://dawn.cs.stanford.edu/benchmark/CIFAR10/train.html) where it obtained 94% accuracy in just 10.7 seconds which is the current best score on 4 GPU and second fastest overall. Additionally, Mish has shown to improve convergence rate by requiring less epochs. Reference - 

![0 (2)](https://user-images.githubusercontent.com/34192716/72232207-9c0b7180-35e5-11ea-920f-295a7c873216.jpg)

Mish also has shown consistent improved ImageNet scores and is more robust. Reference - 

![0](https://user-images.githubusercontent.com/34192716/72232233-b9d8d680-35e5-11ea-97bb-fdb715a6f276.jpg)

Additional ImageNet benchmarks along with Network architectures and weights are avilable on my [repository](https://github.com/digantamisra98/Mish).

Summary of Vision related results: 

![Capture](https://user-images.githubusercontent.com/34192716/72232240-c78e5c00-35e5-11ea-9d1e-dd69f1ef33a7.PNG)

It would be nice to have Mish as an option within the activation function group.

This is the comparison of Mish with other conventional activation functions in a SEResNet-50 for CIFAR-10: 
![se50_1](https://user-images.githubusercontent.com/34192716/69002745-0de37980-091b-11ea-87da-ac8d17c79e07.png)

**Comment by keerthanvasist** [comment]
Thank you for the feature request @digantamisra98. We will implement the Mish activation function soon.

**Comment by digantamisra98** [comment]
@keerthanvasist Thank you for considering the request.

**Comment by digantamisra98** [comment]
Noticed, that Mish has been recently committed to DJL - https://github.com/awslabs/djl/commit/b06feb110e5100db2f839c80a8aca03ba0ecbdcf
Closing this issue then. Thanks for adding Mish.
